### ISSUE TITLE
`RC_TIMESTAMP_MODE` explained

### ISSUE THREAD

**Issue by matanui159** [cisco/openh264#3417]
I have been playing around with OpenH264 encoder settings to try and get a nice balance between quality and file size, and I recently noticed that setting the RC mode to `RC_TIMESTAMP_MODE` seems to make it better at hitting the bitrate target (without frame-skip enabled) while not sacrificing too much quality.

However, I do not understand what `RC_TIMESTAMP_MODE` means and how timestamps have to do with better bitrate control, and I can't find much info on it nor can I understand the code. Can I get a quick ELI5 as to what this does? What are the benefits and what will I be sacrificing to use this? Is there a reason its not used by default?

**Comment by kanzhu-cisco** [comment]
RC_TIMESTAMP_MODE is a kind of rate control mode, which allocate target bits more strictly based on timestamp to get a better CBR result.

**Comment by matanui159** [comment]
So `RC_TIMESTAMP_MODE` is more of a CBR-style bitrate control? While, something like `RC_QUALITY_MODE` would be more VBR? How does using the timestamp lead to better results?

Most of this is more out of curiosity but still I'd like to know if there is some tradeoff that is happening that I do not know of. For instance, if it is better why wouldn't it be default? Compatibility?

**Comment by kanzhu-cisco** [comment]
The frame-level target-bits and quantization is related to frame complexity which weighted accumulated from already encoded frames. "RC_TIMESTAMP_MODE" can get a more accurate target bits than other method like "buffer status".
There is no absolute best rate control method, the choice of strategy depends on you application scenario, in other words, what's the most important to you, the video quality, the stream size or the frame size fluctuation.
